vizro-core/examples/features/default/app.py

```diff
@@ -0,0 +1,428 @@
+)
+
+
+# CUSTOM CHARTS ------------------------------------------------------------------
+@capture("graph")
+def scatter_with_line(data_frame, x, y, hline=None, title=None):
+    """Custom scatter chart based on px."""
+    fig = px.scatter(data_frame=data_frame, x=x, y=y, title=title)
+    fig.add_hline(y=hline, line_color="orange")
+    return fig
+
+
+@capture("graph")
+def waterfall(data_frame, measure, x, y, text, title=None):
+    """Custom waterfall chart based on go."""
+    fig = go.Figure()
+    fig.add_traces(
+        go.Waterfall(
+            measure=data_frame[measure],
+            x=data_frame[x],
+            y=data_frame[y],
+            text=data_frame[text],
+            decreasing={"marker": {"color": "#ff5267"}},
+            increasing={"marker": {"color": "#08bdba"}},
+            totals={"marker": {"color": "#00b4ff"}},
+        ),
+    )
+    fig.update_layout(title=title)
+    return fig
+
+
+custom_charts = vm.Page(
+    title="Custom Charts",
+    components=[
+        vm.Graph(
+            id="custom_scatter",
+            figure=scatter_with_line(
+                x="sepal_length",
+                y="sepal_width",
+                hline=3.5,
+                data_frame=iris,
+                title="Custom px chart",
+            ),
+        ),
+        vm.Graph(
+            id="custom_waterfall",
+            figure=waterfall(
+                data_frame=waterfall_df, measure="measure", x="x", y="y", text="text", title="Custom go chart"
+            ),
+        ),
+    ],
+    controls=[
+        vm.Filter(column="petal_width", targets=["custom_scatter"]),
+        vm.Filter(
+            column="x", targets=["custom_waterfall"], selector=vm.Dropdown(title="Financial categories", multi=True)
+        ),
+    ],
+)
+
+
+# CUSTOM COMPONENTS -------------------------------------------------------------
+# 1. Extend existing components
+class TooltipNonCrossRangeSlider(vm.RangeSlider):
+    """Custom numeric multi-selector `TooltipNonCrossRangeSlider`."""
+
+    type: Literal["other_range_slider"] = "other_range_slider"
+
+    def build(self):
+        """Extend existing component by calling the super build and update properties."""
+        range_slider_build_obj = super().build()
+        range_slider_build_obj[self.id].allowCross = False
+        range_slider_build_obj[self.id].tooltip = {"always_visible": True, "placement": "bottom"}
+        return range_slider_build_obj
+
+
+vm.Filter.add_type("selector", TooltipNonCrossRangeSlider)
+
+
+# 2. Create new custom component
+class Jumbotron(vm.VizroBaseModel):
+    """New custom component `Jumbotron`."""
+
+    type: Literal["jumbotron"] = "jumbotron"
+    title: str
+    subtitle: str
+    text: str
+
+    def build(self):
+        """Build the new component based on Dash components."""
+        return html.Div(
+            [
+                html.H2(self.title),
+                html.H3(self.subtitle),
+                html.P(self.text),
+            ]
```

Comment:
Setting ID for new custom component seems useful concept for users to understand, especially if they need to interact with this component by using custom actions. We should probably add it in the docs as well.  
```suggestion
        return html.Div(
            id=self.id,
            [
                html.H2(self.title),
                html.H3(self.subtitle),
                html.P(self.text),
            ]
```